lib/common/ns.c

```diff
@@ -175,6 +223,12 @@ void init_rank(void)
     queue_free(&Q);
 }
 
+// Return an edge in the feasible tree with a negative cut value, otherwise NULL.
+// 
+// S_i is the index to start the search.  If not found to the end, loop around to
+// zero and search up to S_i.  This is to support a finding in the paper that
```

Comment:
nit: instead of saying “the paper” it might be better to mention the exact paper being referred to here.